src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -191,8 +192,15 @@ public static LocalDate parseDate(String date) throws ParseException {
      * @throws ParseException to be thrown if there is error.
      */
     public static void checkDateExist(LocalDateTime date, String day) throws ParseException {
+        boolean isLeapYear = date.getYear() % 4 == 0;
```

Comment:
Might want to add a few extra conditions for this:
```
A year may be a leap year if it is evenly divisible by 4. 
Years that are divisible by 100 (century years such as 1900 or 2000) cannot be leap years unless they are also divisible by 400. 
(For this reason, the years 1700, 1800, and 1900 were not leap years, but the years 1600 and 2000 were.)
```
The above was taken from Google